src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/ApplicationSettingsController.cs

```diff
@@ -27,14 +37,15 @@ public ApplicationSettingsController(IOptions<AppSettings> appSettings)
         [HttpGet("{org}/{app}/api/v1/applicationsettings")]
         public IActionResult GetAction(string org, string app)
         {
-            return Ok(GetSettings());
-        }
+            FrontEndSettings frontEndSettings = _frontEndSettings;
 
-        private SimpleAppSettings GetSettings()
-        {
-            SimpleAppSettings settings = new SimpleAppSettings();
-            settings.AppOidcProvider = _appSettings.AppOidcProvider;
-            return settings;
+            // Adding key from _appSettings to be backwards compatible.
```

Comment:
We have a single app in TT02, developed by ourselves using this functionality. Other than a one liner in the changelog, how to use this functionality has not been documented. Do we really need the backward compatibility here?

Reply:
It might be possible to remove the compatibility for the applicationsettings endpoint in isolation, but there are more code that still expect to find the AppOidcProvider value under AppSettings. This means I would need to duplicate the value across AppSettings and FrontEndSettings.

I could choose to also change HomeController and have the logic there rely on FrontEndSettings instead of AppSettings. (I don't think the rabbit hole goes any deeper than that.) That might be fine, but I would prefer back end code to use type safe properties to a dictionary of settings.